File: backend/app/services/audit.py
from sqlalchemy.orm import Session

from app.models.audit_log import AuditLog


def write_audit_log(
    db: Session,
    *,
    actor_user_id: int | None,
    action: str,
    target_type: str | None = None,
    target_id: str | None = None,
    ip: str | None = None,
    user_agent: str | None = None,
    success: bool = True,
    detail: str | None = None,
) -> None:
    db.add(
        AuditLog(
            actor_user_id=actor_user_id,
            action=action,
            target_type=target_type,
            target_id=target_id,
            ip=ip,
            user_agent=user_agent,
            success=success,
            detail=detail,
        )
    )

File: frontend/src/types/audit.ts
export type AuditLogItem = {
  id: number
  actor_user_id?: number | null
  actor_username?: string | null
  action: string
  target_type?: string | null
  target_id?: string | null
  ip?: string | null
  user_agent?: string | null
  success: boolean
  detail?: string | null
  created_at: string
}

export type AuditLogListResponse = {
  page: number
  page_size: number
  total: number
  items: AuditLogItem[]
}

